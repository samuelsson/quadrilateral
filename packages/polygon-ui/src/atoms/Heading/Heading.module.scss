@mixin base {
  margin-top: 2.5rem;
  margin-bottom: 1rem;
}

.H1 {
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
  font-size: 1.8rem;
}

.H2 {
  @include base;
  font-size: 1.4rem;
}

.H3 {
  @include base;
  font-size: 1.2rem;
}

.H4 {
  @include base;
  font-size: 1.1rem;
}

.H5 {
  @include base;
  font-size: 1rem;
}

.H6 {
  @include base;
  font-size: 1rem;
}

.Center {
  text-align: center;
}
