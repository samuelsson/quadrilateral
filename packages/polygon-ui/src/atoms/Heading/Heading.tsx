import React from 'react';
import cn from 'classnames';
import styles from './Heading.module.scss';

export interface HeadingProps {
  children: React.ReactNode;
  center?: boolean;
}

export const H1 = ({ children, center }: HeadingProps): JSX.Element => (
  <h1 className={cn(styles.H1, center && styles.Center)}>{children}</h1>
);

export const H2 = ({ children, center }: HeadingProps): JSX.Element => (
  <h2 className={cn(styles.H2, center && styles.Center)}>{children}</h2>
);

export const H3 = ({ children, center }: HeadingProps): JSX.Element => (
  <h3 className={cn(styles.H3, center && styles.Center)}>{children}</h3>
);

export const H4 = ({ children, center }: HeadingProps): JSX.Element => (
  <h4 className={cn(styles.H4, center && styles.Center)}>{children}</h4>
);

export const H5 = ({ children, center }: HeadingProps): JSX.Element => (
  <h5 className={cn(styles.H5, center && styles.Center)}>{children}</h5>
);

export const H6 = ({ children, center }: HeadingProps): JSX.Element => (
  <h6 className={cn(styles.H6, center && styles.Center)}>{children}</h6>
);
