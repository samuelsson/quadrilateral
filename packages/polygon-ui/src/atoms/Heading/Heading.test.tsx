import * as React from 'react';
import { render } from '@testing-library/react';
import { H1 } from './Heading';

describe('Input', (): void => {
  it('renders with no errors', (): void => {
    const { asFragment } = render(<H1>Heading</H1>);

    expect(asFragment()).toMatchSnapshot();
  });
});
