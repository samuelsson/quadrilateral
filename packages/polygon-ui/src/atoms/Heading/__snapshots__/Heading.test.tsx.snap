// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders with no errors 1`] = `
<DocumentFragment>
  <h1
    class="H1"
  >
    Heading
  </h1>
</DocumentFragment>
`;
